api/app/models/retro.rb

```diff
@@ -97,4 +115,16 @@ def generate_slug
 
     slug
   end
+
+  def should_recompute_join_token?
+    magic_link_enabled? && :will_save_change_to_encrypted_password?
```

Comment:
```suggestion
    magic_link_enabled? && will_save_change_to_encrypted_password?
```

The current bit works as is, but as a weird side effect that `false && :will_save_change_to_encrypted_password?` returns a symbol (cuz of default return in ruby) that then gets evaluated in the lifecycle.  Probably better to just call the method directly so you are comparing two booleans and returning a boolean.